cads_adaptors/mapping.py

```diff
@@ -90,18 +94,23 @@ def date_to_julian(ddate):
 
 
 def parse_date(date):
-    if isinstance(date, str):
-        if "-" in date:
+    try:
+        if isinstance(date, str) and "-" in date:
             y, m, d = date.split("-")
-            return int(y) * 10000 + int(m) * 100 + int(d)
-    return int(date)
+            output = int(y) * 10000 + int(m) * 100 + int(d)
+        else:
+            output = int(date)
+    except Exception:
+        raise ValueError(
+            f'Invalid date string: "{date}". Should be ' "yyymmdd or yyyy-mm-dd"
+        )
+    return output
```

Comment:
This bit of code is a bit scary - there's a bit too much wrapped inside the try block, and constructing an int date by multiplyimg all the components like that is a bit weird in my brain!

If we have `dateutil` available, you could do something equivalent like this:
```
def parse_date(date):
    dt = dateutil.parser.parse(date)
    return int(f"{dt:%Y%m%d}")
```

Reply:
I agree, but this leads to bigger changes that I think should addressed in another issue

Reply:
https://github.com/ecmwf-projects/cads-adaptors/issues/133